[DingBao-sys] iP

DingBao-sys opened the pull request:
# ChinesePoliceman
> "I got a ball and I bounce it, I shoot I missed it, SIKE I swished it

You probably will not use it after this module but still 

- here is the bullet list asked for
+ this is nothing compared to chatgpt

If you want to see it you can download it here:

1. First you gotta download it [here](https://github.com/DingBao-sys/ip/tree/A-CodingStandard/src/main/java)
2. Then play with it
3. **Do everything that makes you _happy_** 🦖 

```java
public String getMyCodeBlock() {
    return "yeeeaaaaaa";
}
```

Let me experiment with inline code, I wrote this method ```getMyCodeBlock()```

#TaskList added this to see the heading tab

- [X] #38 
- [ ] #398

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

DingBao-sys committed:
Finished Level-0

DingBao-sys committed:
added .gitignore

DingBao-sys committed:
Level-1

DingBao-sys committed:
Level-2

DingBao-sys committed:
Level-3-Mark-as-Done

DingBao-sys committed:
Level-4-ToDo-Events-Deadline

DingBao-sys committed:
Level-6-Exceptions

DingBao-sys committed:
Level-6-Delete

DingBao-sys committed:
Delete src/main/java/Command.java

DingBao-sys committed:
level-6

DingBao-sys committed:
A-TextUiTesting

DingBao-sys committed:
Done Enums

DingBao-sys committed:
Complete Level 7, Add Save

DingBao-sys committed:
Merge branch 'branch-Level-7'
Current Duke do not support data persistence

Adding Level-7 Save allows user to save task even after duke reloads

DingBao-sys committed:
halfway through lvl 8

DingBao-sys committed:
halfway through OOP

DingBao-sys committed:
Completed branch-level-8

DingBao-sys committed:
Merge branch 'branch-level-8'
Allow event and deadlines to undrstand dates as proper

DingBao-sys committed:
Merge branch 'branch-Level-8' into A-MoreOOP
to incorporate dates and time level

DingBao-sys committed:
Finished A-OOP

DingBao-sys committed:
Merge branch 'A-MoreOOP'
"Made all classes secular"

DingBao-sys committed:
Completed A-Packages

DingBao-sys committed:
Delete src/main/java/ByeCommand.java

DingBao-sys committed:
Delete src/main/java/main/java/Duke/util directory

DingBao-sys committed:
Delete src/main/java/Commands.java

DingBao-sys committed:
Complete A-Packages

DingBao-sys committed:
Merge branch 'A-Packages'
"Divided all files into packages"

DingBao-sys committed:
pull gradle remote branch

DingBao-sys committed:
A-Gradle

DingBao-sys committed:
Merge branch 'A-Gradle'
Incorporated Gradle into Porject

DingBao-sys committed:
Finally Completed Level 9

DingBao-sys committed:
Merge branch 'branch-Level-9'
Add Find Command

DingBao-sys committed:
Complete A-JUnit

DingBao-sys committed:
Merge branch 'A-JUnit'
"add test for tasks and commands class"

smolegz commented on src/main/java/Duke/main/Duke.java:
```diff
@@ -0,0 +1,74 @@
+    private final TaskList tasks;
+    private static Commands parseCommand(String s) throws IllegalArgumentException {
+        String[] words = s.split(" ", 2);
+        Commands res = null;
```
Would it be clearer if 'res' was renamed to 'result'?

smolegz commented on src/main/java/Duke/main/Main.java:
```diff
@@ -0,0 +1,26 @@
+import java.nio.file.Path;
+import java.nio.file.Files;
+import java.io.IOException;
+public class Main {
```
Missed an empty line between import statement and main method

smolegz commented on src/main/java/Duke/tasks/Deadline.java:
```diff
@@ -0,0 +1,82 @@
+            this.deadline = deadline;
+        }
+    }
+    public Deadline(String status, String descr, String deadline) {
```
Would it be better if 'descr' was renamed to 'description' since 'descr' is rather hard to pronounce?

smolegz commented on src/main/java/Duke/tasks/Task.java:
```diff
@@ -0,0 +1,35 @@
+    private final String description;
+    private String status = "[ ]";
+    public abstract boolean hasDate(LocalDateTime toFind);
+    public String writeObject() {
```
Do you think this method would be more explicit if it was named as encodeTask(), since the return type (in this case String) of this method is going to be stored in a file?

smolegz commented on src/main/java/Duke/util/Storage.java:
```diff
@@ -0,0 +1,67 @@
+            while (token.hasNextLine()) {
+                String line = token.nextLine();
+                String[] splits = line.split(" [|] ");
+                Task readTask = null;
```
Can be shortened to:

```Java
Task readTask;
```

smolegz commented on src/main/java/Duke/util/TaskList.java:
```diff
@@ -0,0 +1,46 @@
+        items.add(t);
+        return t;
+    }
+    public ArrayList<Task> list() {
```
Since it is a getter method, naming it getList() would standardize to your other method names :smile:.

smolegz reviewed (commented):
Codebase is well-structured with properly defined classes, and method implementations were simple to digest and understand. Cheers!

DingBao-sys committed:
Include javadoc for all comments

DingBao-sys committed:
Merge branch 'A-JavaDoc'
Included JavaDoc for most methods

DingBao-sys committed:
Tweak to adhere to javadoc

DingBao-sys committed:
Merge branch 'A-CodingStandard'
to be a good responsible coder that codes with standards

DingBao-sys committed:
Adhere to CheckStyle

DingBao-sys committed:
merged with A-Checkstyle

DingBao-sys committed:
finished MainWindow UI

DingBao-sys committed:
non buggy ui

DingBao-sys committed:
Duke classes: incorporate JavaFX GUI

Duke ChatBot uses an interactive and boring textUI which makes in less
compelling for uses to want to use it.

Lets use JavaFX to make the GUI of Duke to be more interactive and add
the much needed vibrancy to compel users to use Duke.
Lets add a viewport for users to see their messages and the replies
returned by duke.
Allow users to interact with duk with a text area and a button to send
their inputs

JavaFX is used as it offers much support with Java which Duke is build
on. Furthermore, in the context of 2103T much support is provided by
the previous batch that used t.

Command Classes: Extracted each Command to have their own classes

The behaviors of the different command is inline with the execution
of Duke run() method.

While the commands have similar behaviors, each command still have
their different purpose. Therefore, code refactoring is done to
seperate them into classes to achieve:
        1. SLAP HARD
        2. DRY(avoiding code duplication)
        3. Scalability

Created an abstract Command class and have the different commands
inherit from the command class and to define their own behavior
in the execute method

Using inheritance allow for the methods in Duke to be more flexible
and use polymorphism to have their own behaviors

DingBao-sys committed:
Duke classes: incorporate JavaFX GUI

    Duke ChatBot uses an interactive and boring textUI which makes in less
    compelling for uses to want to use it.

    Lets use JavaFX to make the GUI of Duke to be more interactive and add
    the much needed vibrancy to compel users to use Duke.
    Lets add a viewport for users to see their messages and the replies
    returned by duke.
    Allow users to interact with duk with a text area and a button to send
    their inputs

    JavaFX is used as it offers much support with Java which Duke is build
    on. Furthermore, in the context of 2103T much support is provided by
    the previous batch that used t.

    Command Classes: Extracted each Command to have their own classes

    The behaviors of the different command is inline with the execution
    of Duke run() method.

    While the commands have similar behaviors, each command still have
    their different purpose. Therefore, code refactoring is done to
    seperate them into classes to achieve:
            1. SLAP HARD
            2. DRY(avoiding code duplication)
            3. Scalability

    Created an abstract Command class and have the different commands
    inherit from the command class and to define their own behavior
    in the execute method

    Using inheritance allow for the methods in Duke to be more flexible
    and use polymorphism to have their own behaviors

DingBao-sys committed:
settle code quality

DingBao-sys committed:
Merge pull request #2 from DingBao-sys/branch-A-CodeQuality

settle code quality

DingBao-sys committed:
Follow commit message conventions

DingBao-sys committed:
Merge pull request #3 from DingBao-sys/Git-Standard

Follow commit message conventions

DingBao-sys committed:
insert assertions for important assumptions

DingBao-sys committed:
add assert statements

DingBao-sys committed:
reolve merge conflict

DingBao-sys committed:
Merge pull request #4 from DingBao-sys/branch-A-Assertions

Branch-a-assertions

DingBao-sys committed:
include BCD-features
        1. User task list will now always be sorted in terms of earliest
           date to be completed. Therefore, todos with no definite
           complete time will always be placed at the back of the list
        2. Added clear command so that users will now be able to clear
           their list

DingBao-sys committed:
add clear to the list of commands

DingBao-sys committed:
Merge branch 'BCD-Extension'
Bring change from BCD branch to master branch for commit

DingBao-sys committed:
hit UI.png

DingBao-sys committed:
Add User guide in ReadMe.md file

DingBao-sys committed:
Edit MarkDown file to remove incompatible emojis in file

edit list messages to have a different empty list message

DingBao-sys committed:
correct name fo Ui.png
